ci/azure-pipelines-dev.yml

```diff
@@ -144,18 +144,23 @@ stages:
 
     jobs:
     - job: static_type_check
+      strategy:
+        matrix:
+          Python38:
+            python.version: '3.8'
```

Comment:
Do we need a matrix if we're just using a single version? Could we just set the `versionSpec` with a hardcoded `3.8` or use Azure variables: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch ?

Reply:
Good catch - originally I had this stage running 3.7 and 3.8. I'll clean it up.

Reply:
Would 3.8, 3.9 be useful here?

Reply:
Changed to a variable in https://github.com/great-expectations/great_expectations/pull/7647. @Shinnnyshinshin - I think we can go to 3.10 but I'm not sure how big of a lift that will be. I'll check on that separately.